packages/flame_tiled/lib/src/tile_atlas.dart

```diff
@@ -95,9 +95,8 @@ class TiledAtlas {
       // The map contains one image, so its either an atlas already, or a
       // really boring map.
       final tiledImage = imageList.first;
-      final image =
-          (await Flame.images.load(tiledImage.source!, key: key)).clone();
-
+      final image = (await Flame.images.load(tiledImage.source!)).clone();
+      Flame.images.add(key, image);
```

Comment:
There will also be two image objects added to the cache now right?

Reply:
Yes, you are right. But they share an image reference.
At present, `TiledAtlas` has a rule for storing keys.
 
 1. No image: `atlas{empty}`
 2. A image: `atlas{$source}`
 3. Images: `atlas{$source1,$source2}`

To keep this rule, I added this: `Flame.images.add(key, image);`.
But If there is no need to keep this rule, the line can be eliminated.

I prefer removing this but it's totally on your call.

Reply:
Since the image is cloned before it is added to the cache again they don't share the reference.
I don't think there is a need to keep that rule, what do you think? :)

Reply:
> Since the image is cloned before it is added to the cache again they don't share the reference.
Oh, I was totally missed it. Yes, you are right again!
> I don't think there is a need to keep that rule, what do you think? :)
I absolutely agree with you since there is a `key` property to `TiledAtlas` so that it could get a cached image if anyone wanted.

Reply:
Also, do you think the `clone` is needed?

Reply:
Not sure, feels like there must have been a reason for someone to put the clone there? Can you check who did it and we can ask? I'm currently on my phone.

Reply:
Sure.

Hi @jtmcdole, can I ask a reason for the `clone` exists? I could roughly assume but curious about the exact reason.

Reply:
@spydon Maybe he's unavailable now. We should leave the `clone` and ask him later :)

Reply:
You can keep using the image so long as it is in the cache. We don't go back to the cache each time we want to reference the image and instead use ` final Image? atlas;`.  If the cache is cleared for any reason, it disposes of the image and we now have an invalid image.

I remember there being a crash due to this... there **should** have been a test as well :)

Reply:
Probably some of your use cases are evicting the cache. Thanks for the answer 😄